src/index.ts

```diff
@@ -137,14 +138,23 @@ const recurseRelationships = (
 					const referencedInterface = modelNameToCamelCaseName(
 						referencedTable.name,
 					);
-					return `${parentKey}?: ${referencedInterface}[];`;
+					// iff the constructed property name matches wuth the full fact type,
+					// replace it with the fact type if there is one.
+					const normalizedParentKey = parentKey
+						.replace(/__/g, '-')
+						.replace(/_/g, ' ');
+					const propName =
+						inverseSynonyms[normalizedParentKey]?.replace(/ /g, '_') ??
+						parentKey;
```

Comment:
I originally tried to not rely on the `parentKey` and do it like:
```ts
const propName =
	referencedField[1] === currentTable.name &&
	referencedTable.name !== referencedTable.resourceName
		? referencedTable.name.replace(/ /g, '_')
		: parentKey;
```
but then I realized that we are emitting more than one property in cases like:
```ts
export interface DeviceType {
	// ...
	device_type__is_referenced_by__alias?: DeviceTypeAlias[];
	is_referenced_by__alias?: DeviceTypeAlias[];
	// ...
}
```
and following that approach would replace both with `device_type_alias`.
As a result I realized I ended up using the `parentKey` so that the prop name is substituted only when it matches the full fact type.

Reply:
@Page- let me then change this to just add the term form as well and switch it to a minor.